nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/FuzzyHashContent.java

```diff
@@ -0,0 +1,192 @@
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("fuzzyhash.value")
+            .build();
+
+    public static final PropertyDescriptor HASH_ALGORITHM = new PropertyDescriptor.Builder()
+            .name("HASH_ALGORITHM")
+            .displayName("Hashing Algorithm")
+            .description("The hashing algorithm utilised")
+            .allowableValues(allowableValueSSDEEP, allowableValueTLSH)
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("Success")
+            .description("Any FlowFile that is successfully hashed will be sent to this Relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("Failure")
+            .description("Any FlowFile that is successfully hashed will be sent to this Relationship.")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(ATTRIBUTE_NAME);
+        descriptors.add(HASH_ALGORITHM);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final ComponentLog logger = getLogger();
+        final String algorithm = context.getProperty(HASH_ALGORITHM).getValue();
+
+
+        final AtomicReference<String> hashValueHolder = new AtomicReference<>(null);
+
+        try {
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream in) throws IOException {
+                    try (ByteArrayOutputStream holder = new ByteArrayOutputStream()) {
+                        StreamUtils.copy(in,holder);
+
+                        if (algorithm.equals(allowableValueSSDEEP.getValue())) {
```

Comment:
just as a note. I did change the code to equals(allowableValueSSDEEP) but it failed to behave as expected, so I rolled back to a similar approach but without the declaration of the algorithm string.